src/abstractdataframe/abstractdataframe.jl

```diff
@@ -2368,16 +2395,36 @@ function flatten(df::AbstractDataFrame,
         col_to_flatten = df[!, col]
         fast_path = eltype(col_to_flatten) isa AbstractVector &&
                     !isempty(col_to_flatten)
-        flattened_col = fast_path ?
-            reduce(vcat, col_to_flatten) :
-            collect(Iterators.flatten(col_to_flatten))
+        flattened_col = if fast_path
+                reduce(vcat, col_to_flatten)
```

Comment:
Is this really untested?

Reply:
it was 😢 , because we had `eltype(col_to_flatten) isa AbstractVector` and it should be `eltype(col_to_flatten) <: AbstractVector`. Changed. Thank you for spotting.